applications/nrf_desktop/src/modules/led_stream.c

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+
+#define MODULE led_stream
+#include "module_state_event.h"
+
+#include <logging/log.h>
+
+#include "config_event.h"
+#include "led_event.h"
+#include "led_ready_event.h"
+
+#define STREAM_LED_ID 0
+#define QUEUE_SIZE 5
+
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_LED_STREAM_LOG_LEVEL);
+
+static struct led_effect led_stream_effect = {
+	.step_count = 1,
+};
+static struct led_effect_step steps_queue[QUEUE_SIZE];
+static s8_t last_added_position = -1;
+static s8_t send_position;
+static u8_t free_places = QUEUE_SIZE;
+static bool first_sent;
+static bool leds_initialized;
+
+static void queue_data(const struct event_dyndata *dyndata)
+{
+	LOG_INF("Insert_data_to_effect");
+	BUILD_ASSERT(CONFIG_DESKTOP_LED_COUNT == 3);
```

Comment:
we need to accept lower led count - consider passing index of led to which stream is sent